R/carehomes.R

```diff
@@ -872,31 +874,37 @@ carehomes_parameters_strain <- function(strain_transmission, strain_seed_date,
   }
 
   if (is.null(strain_seed_date)) {
-    if (!is.null(strain_seed_value)) {
-      stop(paste("As 'strain_seed_date' is NULL, expected 'strain_seed_value'",
+    if (!is.null(strain_seed_rate)) {
+      stop(paste("As 'strain_seed_date' is NULL, expected 'strain_seed_rate'",
                  "to be NULL"))
     }
     strain_seed_step <- 0
   } else {
     if (length(strain_transmission) == 1L) {
       stop("Can't use 'strain_seed_date' if only using one strain")
     }
-    if (length(strain_seed_date) != 2L) {
-      stop("'strain_seed_date', if given, must be exactly two elements")
+    if (length(strain_seed_date) != length(strain_seed_rate)) {
+      stop("'strain_seed_date' and 'strain_seed_rate' must be the same length")
     }
     assert_sircovid_date(strain_seed_date)
     assert_increasing(strain_seed_date, strict = FALSE)
 
-    if (length(strain_seed_value) != 1L) {
-      stop("'strain_seed_value' must be a scalar if 'strain_seed_date' is used")
+    ## The + 1 here prevents the start of the next day having the
+    ## same seeding value
+    strain_seed_step <-
+      numeric(strain_seed_date[[length(strain_seed_date)]] / dt)
+    for (j in seq_along(strain_seed_date)) {
+      if (j == length(strain_seed_date)) {
+        i <- length(strain_seed_step)
+      } else {
+        i <- seq.int(
+          strain_seed_date[[j]] / dt,
+          strain_seed_date[[j + 1]] / dt - 1
+        )
+      }
+      strain_seed_step[i] <- strain_seed_rate[[j]] * dt
     }
-    assert_integer(strain_seed_value)
```

Comment:
we should probably replace this by a check that the values in there are all positive?

Reply:
and add corresponding tests